youtube_dl/extractor/vxxx.py

```diff
@@ -0,0 +1,240 @@
+            - "." is used instead of "+"; "," is used instead of "/"
+            - "~" is used for padding instead of "="
+        """
+
+        # using the kwarg to memoise the result
+        def get_trans_tbl(from_, to, tbl={}):
+            k = (from_, to)
+            if not tbl.get(k):
+                tbl[k] = str.maketrans(from_, to)
+            return tbl[k]
+
+        trans_tbl = get_trans_tbl(
+            '\u0410\u0412\u0421\u0415\u041c.,~',
+            'ABCEM+/=')
+        return base64.b64decode(e.translate(trans_tbl)).decode()
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        info_object = self._download_info_object(video_id)
+
+        title = info_object['title']
+        stats = info_object.get('statistics') or {}
+        info = {
+            'id': video_id,
+            'title': title,
+            'display_id': info_object.get('dir'),
+            'thumbnail': url_or_none(info_object.get('thumb')),
+            'description': strip_or_none(info_object.get('description')) or None,
+            'timestamp': unified_timestamp(info_object.get('post_date')),
+            'duration': parse_duration(info_object.get('duration')),
+            'view_count': int_or_none(stats.get('viewed')),
+            'like_count': int_or_none(stats.get('likes')),
+            'dislike_count': int_or_none(stats.get('dislikes')),
+            'average_rating': float_or_none(stats.get('rating')),
+            'categories': [category['title'] for category in (info_object.get('categories') or {}).values()
+                           if category.get('title')],
+            'age_limit': 18,
+        }
+
+        format_object = self._download_format_object(video_id)
+        m3u8_formats = self._extract_m3u8_formats(
+            '{0}/{1}&f=video.m3u8'.format(
+                self._BASE_URL,
+                self._decode_base164(format_object[0]['video_url'])
+            ),
+            video_id, 'mp4')
+        self._sort_formats(m3u8_formats)
+        info['formats'] = m3u8_formats
+
+        return info
+
+
+class InPornIE(VXXXIE):
+    _VALID_URL = r'https?://(?:www\.)?inporn\.com/video/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://inporn.com/video/533613/2k-t-2nd-season-parm-151/',
+        'md5': 'c358d1da6b451ebe7cfb00dd89741607',
+        'info_dict': {
+            'id': '533613',
+            'ext': 'mp4',
+            'title': '2k 美月まい - ガーリー系アパレルモt゙ルの挑発パンチラ 2nd Season [parm-151]',
+            'display_id': '2k-t-2nd-season-parm-151',
+            'thumbnail': 'https://tn.inporn.com/media/tn/533613_1.jpg',
+            'description': None,
+            'timestamp': 1664571262,
+            'upload_date': '20220930',
+            'duration': 480.0,
+            'categories': ['Asian', 'Brunette', 'Casting', 'HD', 'Japanese',
+                           'JAV Uncensored'],
+            'age_limit': 18,
+        },
```

Comment:
If `m3u8_native` doesn't work, put this in (here and in the other tests). The `skip_download` line can be commented out for local testing.
```suggestion
        },
        'params': {
            # ffmpeg download
            'skip_download': True,
        },
```